File: 大二下/算法分析和设计/CCode/6.迭代法/jeep.cpp
#include<iostream>
#include<cstdio>

using namespace std;

void jeep(){
	int k = 1;
	int oil = 500;
	int dis = 500;
	do{
		printf("储油点 %d 距出发点距离 %d 储油量 %d\n",k,1000-dis,oil);
		k++; // 从第二个站点开始计算
		dis = dis + 500/(2*k-1); // 计算站点距离
		printf("%d\n",dis);
		oil = 500 * k; // 计算储油量
	}while(dis < 1000); // 
	oil = 500*(k-1)+(1000-dis)*(2*k-1);
	printf("%d\n",oil);
	return;
}

int main()
{
	jeep();
	return 0;
}

File: 大二下/算法分析和设计/JavaCode/src/第6章_迭代法/jeep.java
package 第6章_迭代法;

/**
 * 6.3.2 输油问题
 */
public class jeep {
    public static void main(String[] args) {
        int s=1000;int l=0;int i=0;
        int[] a=new int[100];
        while(l<1000) {
            a[i]=500*(i+1);
            l+=500/(2*(i+1)-1);
            if(l<1000) {
                System.out.println("第"+(i+1)+"个点，"+"距终点"+l+"米"+"储油量"+a[i]+"升");
            }
            else {//控制距终点长度不超出沙漠总长
                System.out.println("第"+(i+1)+"个点，"+"距终点1000米"+"储油量"+a[i]+"升");
            }
            i++;
        }
    }
}
